src/transformers/modeling_utils.py

```diff
@@ -4661,16 +4661,32 @@ def forward(
 
 def unwrap_model(model: nn.Module) -> nn.Module:
     """
-    Recursively unwraps a model from potential containers (as used in distributed training).
+    Recursively unwraps a module and its child sublayers.
 
     Args:
-        model (`torch.nn.Module`): The model to unwrap.
+        model (nn.Module): The model to unwrap.
+
+    Returns:
+        nn.Module: The unwrapped module.
     """
-    # since there could be multiple levels of wrapping, unwrap recursively
-    if hasattr(model, "module"):
-        return unwrap_model(model.module)
-    else:
-        return model
+
+    def recursive_unwrap(module):
+        if hasattr(module, "module"):
+            try:
+                unwrapped_module = recursive_unwrap(getattr(module, "module"))
+            except AttributeError:
+                unwrapped_module = module  # Handle cases where wrapped module is inaccessible
```

Comment:
Could you give an example of when this happens? It seems weird we'd have `hasattr(module, "module")` evaluate as `True` but then we can't do `getattr(module, "module")`

Reply:
Yes, you are right @amyeroberts. It does seem weird, I don't remember why I implemented like this, but thanks for pointing it out. I also can't think of an example. 

I am fixing it.